### PR TITLE
Allow user to specify an ICredentialsLoader 

### LINKED ISSUE #2564: Not able to override ICredentialsLoader
### Microsoft.Identity.Web Library

Microsoft.Identity.Web.TokenAcquisition

### Microsoft.Identity.Web version

2.15.3

### Web app

Not Applicable

### Web API

Not Applicable

### Token cache serialization

Not Applicable

### Description

TokenAcquisition ServiceCollectionExtensions is checking for ICredentialsLoader against ImplementationType. It either needs to check ICredentialsLoader against ServiceType or ImplemenationType against DefaultCredentialLoader

See:

https://github.com/AzureAD/microsoft-identity-web/blob/master/src/Microsoft.Identity.Web.TokenAcquisition/ServiceCollectionExtensions.cs#L48

### Reproduction steps

1. Register a custom ICredentialsLoader
2. Try to use an API that uses the credentials loader

### Error message

_No response_

### Id Web logs

_No response_

### Relevant code snippets

```csharp
NA
```

### Regression

_No response_

### Expected behavior

The provided ICredentialsLoader is used.

### PR DESCRIPTION
**Pull request by keegan-caruso**
# Allow user to specify an ICredentialsLoader 

Fixes #2564

### CODE REVIEW

**Review by jennyf19 — APPROVED** [review]
:shipit:

File: src/Microsoft.Identity.Web.TokenAcquisition/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -45,9 +44,9 @@ public static class ServiceCollectionExtensions
             bool forceSdk = !services.Any(s => s.ServiceType.FullName == "Microsoft.AspNetCore.Authentication.IAuthenticationService");
 #endif
 
-            if (services.FirstOrDefault(s => s.ImplementationType == typeof(ICredentialsLoader)) == null)
+            if (services.FirstOrDefault(s => s.ImplementationType == typeof(DefaultCertificateLoader)) == null)
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web.TokenAcquisition/ServiceCollectionExtensions.cs:47]: wouldn't want to test that there is no implementation of ICredentialLoader?
that is:

```csharp
if (services.FirstOrDefault(s => s.ServiceType== typeof(ICredentialsLoader)) == null)
{
  services.TryAddSingleton<DefaultCertificateLoader>();
}
```

?

**Review by jmprieur — APPROVED** [review]
LGTM
I left a question / suggestion

File: src/Microsoft.Identity.Web.TokenAcquisition/ServiceCollectionExtensions.cs (hunk 1 above)
Comment (keegan-caruso) [reply src/Microsoft.Identity.Web.TokenAcquisition/ServiceCollectionExtensions.cs:47]: That is how I originally wanted to write it 😄 

I went with this as it is more consistent with the rest of the file.

**Pull request merged**